src/ember-template-compiler.ts

```diff
@@ -21,6 +21,7 @@ export interface PreprocessOptions {
   mode?: 'codemod' | 'precompile';
   strictMode?: boolean;
   locals?: string[];
+  localsWithNames?: { [key: string]: string };
```

Comment:
The intent of this type was to reflect the real params that the ember template compiler accepts. We can't just add new stuff here.

It's fine to introduce a new type instead for stuff that we handle entirely in this project and never pass into ember-template-compiler.

Reply:
Thank you, I am going to address this issue.